<?xml version="1.0"?>
<!--
/****************************************************************************
 *   Copyright (c) 2018 Michael Shomin. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in
 *    the documentation and/or other materials provided with the
 *    distribution.
 * 3. Neither the name ATLFlight nor the names of its contributors may be
 *    used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE.
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
 * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
 * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
 * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
 * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 *
 * In addition Supplemental Terms apply.  See the SUPPLEMENTAL file.
 ****************************************************************************/
-->
<launch>
  <arg name="board" default="sdf_tray"/>
  <arg name="vehicle" default="ddk_tray"/>
  <arg name="imu_is_root_tf" default="false"/>

  <arg name="rviz" default="false"/>

  <param name="board_description" textfile="$(find qflight_descriptions)/urdf/$(arg board).urdf"/>
  <node name="board_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <remap from="robot_description" to="board_description"/>
  </node>

  <param name="vehicle_description" textfile="$(find qflight_descriptions)/urdf/$(arg vehicle).urdf"/>
  <node name="vehicle_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <remap from="robot_description" to="vehicle_description"/>
  </node>

  <node name="base_link_imu_static_publisher" pkg="qflight_descriptions"
        type="base_link_imu_static_publisher.py" output="screen">
    <param name="board_mounts" value="$(find qflight_descriptions)/urdf/board_mounts.yaml"/>
    <param name="imu_is_root_tf" type="bool" value="$(arg imu_is_root_tf)"/>
    <param name="board" value="$(arg board)"/>
    <param name="vehicle" value="$(arg vehicle)"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" output="screen" if="$(arg rviz)"
        args="-d $(find qflight_descriptions)/launch/qflight_urdf.rviz"/>
</launch>
